sdk/src/client.rs

```diff
@@ -87,6 +89,71 @@ impl NetworkClient {
         Ok(())
     }
 
+    pub async fn get_proof_status<SC: StarkGenericConfig + Serialize + DeserializeOwned>(
+        &self,
+        proof_id: &str,
+    ) -> Result<(GetProofStatusResponse, Option<SP1ProofWithIO<SC>>)> {
+        let res = self
+            .rpc
+            .get_proof_status(GetProofStatusRequest {
+                proof_id: proof_id.to_string(),
+            })
+            .await?;
+
+        let proof = if res.status() == ProofStatus::ProofFulfilled {
+            let proof = self
+                .http
+                .get(res.proof_url.clone())
+                .send()
+                .await?
+                .bytes()
+                .await?;
+            Some(bincode::deserialize(&proof).expect("Failed to deserialize proof"))
+        } else {
+            None
+        };
+
+        Ok((res, proof))
+    }
```

Comment:
Add doc comments for each of these functions (as they're public-facing).

Reply:
added docs to all of the NetworkClient functions